src/libguac/guacamole/string.h

```diff
@@ -109,6 +109,28 @@ size_t guac_strlcpy(char* restrict dest, const char* restrict src, size_t n);
  */
 size_t guac_strlcat(char* restrict dest, const char* restrict src, size_t n);
 
+/**
+ * Search for the null-terminated string needle in the possibly null-
+ * terminated haystack, looking at no more than len bytes.
+ *
+ * @param haystack
+ *     The string to search. It may or may not be null-terminated. Only the
+ *     first len bytes are searched.
+ *
+ * @param needle
+ *     The string to look for. It must be null-terminated.
+ *
+ * @param len
+ *     The maximum number of bytes to examine in haystack.
+ *
+ * @return
+ *     If needle is the empty string, return haystack. If needle exists in
+ *     haystack, return a pointer to the first instance in haystack. If needle
+ *     does not exist in haystack, return NULL.
+ *
```

Comment:
The convention for `@return` has been to describe the return value as if the "Returns ..." starting part of the sentence is already written, with the remaining documentation written as if that were already there. For example:

https://github.com/apache/guacamole-server/blob/78cbaf27ba09ea5fee6f7869edb9e7513d382690/src/libguac/guacamole/string.h#L187-L191

I think we should do the same here. My suggestion would be something like:

```c
/**
 *...
 * @return
 *     A pointer to the first instance of needle within haystack, or NULL if
 *     needle does not exist in haystack. If needle is the empty string,
 *     haystack is returned.
 */
```

Besides consistency, I think the order is particularly important here, as the edge case of the empty `needle` is noteworthy but probably not the most critical consideration for a developer looking to invoke `guac_strnstr()`.

Reply:
Done